File: source2graph/test/fixtures/java-sample/src/com/example/service/UserService.java
package com.example.service;

import com.example.model.User;
import com.example.repository.UserRepository;
import java.util.List;
import java.util.Optional;

public class UserService {
    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public Optional<User> findUser(Long id) {
        return userRepository.findById(id);
    }

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    public User createUser(String name, String email) {
        User user = new User(null, name, email);
        return userRepository.save(user);
    }

    public boolean validateUser(User user) {
        if (user.getName() == null || user.getName().isBlank()) {
            return false;
        }
        if (user.getEmail() == null || !user.getEmail().contains("@")) {
            return false;
        }
        return true;
    }
}

File: source2graph/test/fixtures/scala-sample/src/com/example/service/UserService.scala
package com.example.service

import com.example.model.User
import com.example.repository.UserRepository
import scala.collection.mutable.ListBuffer

class UserService(val repo: UserRepository) {
  private val cache: ListBuffer[User] = ListBuffer.empty

  def findById(id: Long): Option[User] = {
    repo.findById(id)
  }

  def createUser(name: String, email: String): User = {
    val user = User(0L, name, email)
    repo.save(user)
  }
}

object UserService {
  def apply(repo: UserRepository): UserService = new UserService(repo)
}
